tests/manage/z_cluster/cluster_exp_helpers.py

```diff
@@ -0,0 +1,215 @@
+
+        """
+        if not self.cluster_filled:
+            target_dir_name = "/mnt/cluster_fillup0_" + uuid4().hex
+            mkdir_cmd = "" + "mkdir " + target_dir_name + ""
+            fill_pod.exec_sh_cmd_on_pod(mkdir_cmd, sh="bash")
+            logging.info(f"#### Created the dir {target_dir_name} on pod {fill_pod.name}")
+            tee_cmd = "" + " tee " + target_dir_name + \
+                "/ceph.tar.gz{1..30} < /mnt/ceph.tar.gz >/dev/null &" + ""
+            logging.info(f"#### Executing {tee_cmd} to fill the cluster space from pod {fill_pod.name}")
+            fill_pod.exec_sh_cmd_on_pod(tee_cmd, sh="bash")
+            logging.info(f"#### Executed command {tee_cmd}")
+
+        return True
+
+    def cluster_filler(self):
+        curl_cmd = f""" curl {constants.REMOTE_FILE_URL} --output {constants.FILE_PATH} """
+        logging.info('downloading......')
+        run_cmd(cmd=curl_cmd)
+        logging.info('finished')
+        with ThreadPoolExecutor() as executor:
+            for pod in self.pods_to_fill:
+                executor.submit(downloader, pod, self.namespace)
+                logging.info(f"### initiated downloader for {pod.name}")
+
+        filler_executor = ThreadPoolExecutor()
+        while not self.cluster_filled:
+            for copy_iter in range(self.concurrent_copies):
+                for each_pod in self.pods_to_fill:
+                    self.used_capacity = get_percent_used_capacity()
+                    logging.info(f"### used capacity %age = {self.used_capacity}")
+                    if self.used_capacity <= self.percent_required_filled:
+                        filler_executor.submit(self.filler, each_pod)
+                        logging.info(f"#### Ran copy operation on pod {each_pod.name}. copy_iter # {copy_iter}")
+                    else:
+                        logging.info(f"############ Cluster filled to the expected capacity "
+                                     f"{self.percent_required_filled}"
+                                     )
+                        self.cluster_filled = True
+                        break
+                if self.cluster_filled:
+                    break
+
+        return True
+
+
+class BackgroundOps():
+    EXPANSION_COMPLETED = False
+
+    def wrap(self, func, *args, **kwargs):
+        """
+        Wraps the function to run specific iterations
+
+        Returns:
+            bool : True if function runs successfully
+        """
+        iterations = kwargs.get('iterations', 1)
+        func_name = func.__name__
+        del kwargs['iterations']
+        for i in range(iterations):
+            if BackgroundOps.EXPANSION_COMPLETED:
+                logger.info(f"{func_name}: Done with execution. Stopping the thread. In iteration {i}")
+                return True
+            else:
+                assert func(*args, **kwargs), f'{func_name} failed!'
+                logger.info(f"{func_name}: iteration {i}")
+
+
+def check_nodes_status():
```

Comment:
this function must exist somewhere in the project, please use existing one

Reply:
Searched, got to know, `node.wait_for_nodes_status(
        node_names=None,
        status=constants.NODE_READY,
        timeout=5)` is getting used directly. I wanted to wrap in a function so that I can call in Background ops